### PR TITLE
Validation of package entries metadata

### PR DESCRIPTION
**Pull request by sir-sigurd**
# Description

# TODO

- [x] Unit tests
- [x] <del>Documentation</del> (added in https://github.com/quiltdata/quilt/pull/3336)
- [x] [Changelog](CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

File: docs/advanced-features/workflows.md [hunk 1]
```diff
@@ -548,11 +548,17 @@ workflows:
...
+  myworkflow-3:
+    name: 'My workflow #3'
+    entries_schema: objects-must-have-id
+    description: All objects must have `id` property of string type in their metadata
 schemas:
   must-contain-readme:
     url: s3://bucket/must-contain-readme.json
   must-contain-readme-summarize-at-least-1byte:
     url: s3://bucket/must-contain-readme-summarize-at-least-1byte.json
+  must-contain-readme-summarize-at-least-1byte:
```
Comment (robnewman) [inline docs/advanced-features/workflows.md:560]:
```suggestion
  objects-must-have-id
```

**Review by robnewman — CHANGES_REQUESTED** [review]
Fix the schema name - otherwise all good!

File: docs/advanced-features/workflows.md [hunk 2]
```diff
@@ -548,11 +548,17 @@ workflows:
...
+  myworkflow-3:
+    name: 'My workflow #3'
+    entries_schema: objects-must-have-id
+    description: All objects must have `id` property of string type in their metadata
 schemas:
   must-contain-readme:
     url: s3://bucket/must-contain-readme.json
   must-contain-readme-summarize-at-least-1byte:
     url: s3://bucket/must-contain-readme-summarize-at-least-1byte.json
+  objects-must-have-id
```
Comment (sir-sigurd) [inline docs/advanced-features/workflows.md:560]:
```suggestion
  objects-must-have-id:
```

**Review by robnewman — DISMISSED** [review]

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ Entries inside each section should be ordered by type:
 
 ## Catalog, Lambdas
 !-->
```
Comment (nl0) [inline docs/CHANGELOG.md:16]: we should probably add the "Unreleased" section here

**Review by nl0 — APPROVED** [review]

File: api/python/tests/test_workflows.py [hunk 4]
```diff
@@ -212,26 +212,29 @@ def test_validate_pkg_entries_fail(self, get_pkg_entries_for_validation_mock):
...
             },
             {
                 'logical_key': 'b/a',
                 'size': 1,
+                "meta": {},
             },
             {
                 'logical_key': 'c',
                 'size': 3,
+                "meta": {"test": "test"},
```
Comment (fiskus) [inline api/python/tests/test_workflows.py:238]: `"meta": {"user_meta": {"test": "test"}}`?

To make test closer to real-life scenario?

File: docs/advanced-features/workflows.md [hunk 5]
```diff
@@ -623,6 +629,31 @@ Requires a README
...
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "meta": {
+        "type": "object",
+        "properties": {
+          "id": {
```
Comment (fiskus) [inline docs/advanced-features/workflows.md:644]: I'd test it with

```js
{
  meta: {
    type: 'object',
    properties: {
      user_meta: {
        type: 'object',
        properties: { id: { type: string }},
        required: ['id']
      }
    },
    required: ['user_meta']
  }
}
```

File: api/python/tests/test_workflows.py (hunk 4 above)
Comment (sir-sigurd) [reply api/python/tests/test_workflows.py:238]: This `meta` is `user_meta`.
See https://github.com/quiltdata/quilt/blob/16f25156608668a5495532372cf572f8de8206eb/api/python/quilt3/packages.py#L183-L185

**Review by fiskus — APPROVED** [review]

**Review by nl0 — APPROVED** [review]

**Pull request merged**